api/src/schemas.py

```diff
@@ -291,3 +311,27 @@ def check_sample_files(cls, values):
                 "of the provided prefix/filename_regex combinations."
             )
         return values
+
+
+class ZarrDataset(Dataset):
+    xarray_kwargs: Optional[Dict] = dict()
+    x_dimension: Optional[str]
+    y_dimension: Optional[str]
+    temporal_dimension: Optional[str]
+    reference_system: Optional[int]
+    data_type: Literal[DataType.zarr]
+
+    @validator("discovery_items")
+    def only_one_discover_item(cls, discovery_items):
+        if len(discovery_items) != 1:
+            raise ValueError("Zarr dataset should have exactly one discovery item")
+        if not discovery_items[0].zarr_store:
+            raise ValueError(
+                "Zarr dataset should include zarr_store in its discovery item"
+            )
+        return discovery_items
+
+
+DatasetUnion = Annotated[
+    Union[COGDataset, ZarrDataset], Field(discriminator="data_type")
+]
```

Comment:
Not sure this is needed - we only need `ItemUnion` because datasets can define multiple patterns for item discovery, but we only accept inputs with one dataset at a time. I think `Union[COGDataset, ZarrDataset], Field(discriminator="data_type")`  can replace uses of `DatasetUnion`.